src/commands/account/export_account/mod.rs

```diff
@@ -0,0 +1,227 @@
+    let account_key_pair: crate::transaction_signature_options::AccountKeyPair =
+        serde_json::from_str(&data)
+            .wrap_err_with(|| format!("Error reading data from file: {:?}", &data_path))?;
+    Ok(account_key_pair)
+}
+
+fn get_account_key_pair_data_path(
+    network_config: &crate::config::NetworkConfig,
+    account_id: &near_primitives::types::AccountId,
+    credentials_home_dir: &std::path::Path,
+) -> color_eyre::eyre::Result<std::path::PathBuf> {
+    let check_if_seed_phrase_exists = false;
+    get_account_properties_data_path(
+        network_config,
+        account_id,
+        credentials_home_dir,
+        check_if_seed_phrase_exists,
+    )
+}
+
+pub fn get_account_properties_data_path(
+    network_config: &crate::config::NetworkConfig,
+    account_id: &near_primitives::types::AccountId,
+    credentials_home_dir: &std::path::Path,
+    check_if_seed_phrase_exists: bool,
+) -> color_eyre::eyre::Result<std::path::PathBuf> {
+    let file_name = format!("{}.json", account_id);
+    let mut path = std::path::PathBuf::from(credentials_home_dir);
+
+    let dir_name = network_config.network_name.clone();
+    path.push(&dir_name);
+    path.push(file_name);
+
+    if path.exists() {
+        if !check_if_seed_phrase_exists {
+            return Ok(path);
+        }
+        let data = std::fs::read_to_string(&path).wrap_err("Access key file not found!")?;
+        if serde_json::from_str::<crate::common::KeyPairProperties>(&data).is_ok() {
+            return Ok(path);
+        }
+    }
+
+    let access_key_list = network_config
+        .json_rpc_client()
+        .blocking_call_view_access_key_list(
+            account_id,
+            near_primitives::types::Finality::Final.into(),
+        )
+        .wrap_err_with(|| format!("Failed to fetch access KeyList for {}", account_id))?
+        .access_key_list_view()?;
+    let mut path = std::path::PathBuf::from(credentials_home_dir);
+    path.push(dir_name);
+    path.push(account_id.to_string());
+    let mut data_path = std::path::PathBuf::new();
+    for access_key in access_key_list.keys {
+        let account_public_key = access_key.public_key.to_string();
+        let is_full_access_key: bool = match &access_key.access_key.permission {
+            near_primitives::views::AccessKeyPermissionView::FullAccess => true,
+            near_primitives::views::AccessKeyPermissionView::FunctionCall {
+                allowance: _,
+                receiver_id: _,
+                method_names: _,
+            } => false,
+        };
+        let dir = path
+            .read_dir()
+            .wrap_err("There are no access keys found in the keychain for the account.")?;
+        for entry in dir.flatten() {
+            if entry
+                .path()
+                .file_stem()
+                .unwrap()
+                .to_str()
+                .unwrap()
+                .contains(account_public_key.rsplit(':').next().unwrap())
```

Comment:
Instead of splitting, use replace, and do it on `account_public_key` variable initialization (above) to avoid this operation being executed for each `entry`.

Reply:
Done